File: Lab1/Lab1.cpp
#include <iostream>

using namespace std;

/**
* Gets the length of a segment based on coords
*/
double getLength(double x1, double x2, double y1, double y2) {
    double length = sqrt(pow((x2 - x1), 2) + pow((y2 - y1), 2));
    return length;
}

int main()
{
    double x1, x2, x3, y1, y2, y3;

    // input data
    cout << "x1= "; cin >> x1;
    cout << "y1= "; cin >> y1;
    cout << "x2= "; cin >> x2;
    cout << "y2= "; cin >> y2;
    cout << "x3= "; cin >> x3;
    cout << "y3= "; cin >> y3;

    // declare length of the sides
    double a = getLength(x1, x2, y1, y2), b = getLength(x2, x3, y2, y3), c = getLength(x3, x1, y3, y1);

    double per, halfPer, area;

    per = a + b + c;
    halfPer = per / 2;

    // using Heron's formula
    area = sqrt(halfPer * (halfPer - a) * (halfPer - b) * (halfPer - c));

    printf("Perimeter = %5.3f\nArea = %5.3f\n", per, area);

    return 0;
}

File: lab_1/lab_1.py
from math import sqrt

def get_length(x1, x2, y1, y2):
	"""return the length of the line based on coords"""
	length = sqrt((x2-x1)**2 + (y2-y1)**2)
	return length

def main():
	try:
		x1, y1 = float(input("x1= ")), float(input("y1= "))
		x2, y2 = float(input("x2= ")), float(input("y2= "))
		x3, y3 = float(input("x3= ")), float(input("x3= "))
	except:
		print("Invalid data")
		return

	a, b, c = get_length(x1, x2, y1, y2), get_length(x2, x3, y2, y3), get_length(x3, x1, y3, y1)

	per = a + b + c
	half_per = per/2

	#using Heron's formula
	area = sqrt(half_per*(half_per-a)*(half_per-b)*(half_per-c))

	print(f"Perimeter = {per}\nArea = {area}")

main()


